components/safe_browsing/android/java/src/org/chromium/components/safe_browsing/BraveSafeBrowsingApiHandler.java

```diff
@@ -72,103 +77,140 @@ private void resetDelegate() {
-                        // additional details.
-                        ApiException apiException = (ApiException) e;
-                        if (isDebuggable()) {
-                            Log.d(TAG,
-                                    "Error: "
-                                            + CommonStatusCodes.getStatusCodeString(
-                                                    apiException.getStatusCode())
-                                            + ", code: " + apiException.getStatusCode());
-                        }
-                        if (apiException.getStatusCode() == CommonStatusCodes.API_NOT_CONNECTED) {
-                            // That means that device doesn't have Google Play Services API.
-                            // Delegate is used to turn off safe browsing option as every request is
-                            // delayed when it's turned on and not working
-                            mBraveSafeBrowsingApiHandlerDelegate.turnSafeBrowsingOff();
-                        }
-
-                        // Note: If the status code, apiException.getStatusCode(),
-                        // is SafetyNetStatusCodes.SAFE_BROWSING_API_NOT_INITIALIZED,
-                        // you need to call initSafeBrowsing(). It means either you
-                        // haven't called initSafeBrowsing() before or that it needs
-                        // to be called again due to an internal error.
-                        if (mTriesCount <= 1
-                                && apiException.getStatusCode()
-                                        == SafetyNetStatusCodes.SAFE_BROWSING_API_NOT_INITIALIZED) {
-                            initSafeBrowsing();
-                            startUriLookup(callbackId, uri, threatsOfInterest);
-                        } else {
-                            mObserver.onUrlCheckDone(callbackId, SafeBrowsingResult.TIMEOUT, "{}",
-                                    DEFAULT_CHECK_DELTA);
-                        }
-                    } else {
-                        // A different, unknown type of error occurred.
-                        if (isDebuggable()) {
-                            Log.d(TAG, "Error: " + e.getMessage());
-                        }
-                        mObserver.onUrlCheckDone(
-                                callbackId, SafeBrowsingResult.TIMEOUT, "{}", DEFAULT_CHECK_DELTA);
-                    }
-                    mTriesCount = 0;
-                });
+                            mTriesCount = 0;
+                        });
     }
 
-    @Override
-    public boolean startAllowlistLookup(final String uri, int threatType) {
-        return false;
+    private void warnWhenMoreThanOneThreat(SafeBrowsingResponse sbResponse) {
+        if (sbResponse.getDetectedThreats().size() != 1) {
+            Log.w(TAG, "Unexpected threats count: " + sbResponse.getDetectedThreats().size());
```

Comment:
why do we need to print that in production? Perhaps a good idea to print it in debug mode only?

Reply:
I agree, changed at https://github.com/brave/brave-core/pull/25842/commits/dd4c302cc51dec3f3409b0d5667d8ce055459b5c#diff-fbe6ef23e5c73345c352043cf4cf6e37296db1fd0b59b63036bfd029124187b8R206

thanks